crypto/fipsmodule/cpucap/cpu_intel.c

```diff
@@ -133,28 +133,47 @@ static void handle_cpu_env(uint32_t *out, const char *in) {
   const int or = in[0] == '|';
   const int skip_first_byte = invert || or;
   const int hex = in[skip_first_byte] == '0' && in[skip_first_byte+1] == 'x';
+  uint32_t intelcap0 = out[0];
+  uint32_t intelcap1 = out[1];
 
   int sscanf_result;
   uint64_t v;
   if (hex) {
-    sscanf_result = sscanf(in + invert + 2, "%" PRIx64, &v);
+    sscanf_result = sscanf(in + skip_first_byte + 2, "%" PRIx64, &v);
   } else {
-    sscanf_result = sscanf(in + invert, "%" PRIu64, &v);
+    sscanf_result = sscanf(in + skip_first_byte, "%" PRIu64, &v);
   }
 
   if (!sscanf_result) {
     return;
   }
 
+  uint32_t reqcap0 = (uint32_t)(v & UINT32_MAX);
+  uint32_t reqcap1 = (uint32_t)(v >> 32);
+
+  // Detect if the user is trying to use the environment variable to set
+  // a capability that is _not_ available on the CPU.
+  // The case of invert cannot enable an unexisting capability;
+  // it can only disable an existing one.
+  if (!invert && (intelcap0 || intelcap1)) {
+    // Still allow RDRAND bit to be set for testing
```

Comment:
I think you mean that this reserved bit indicates Intel cpu?

Reply:
Ahh.. you're right!  I update the comment.

Reply:
This is not very clear to me. We could have disabled the RDRAND, e.g. in https://github.com/aws/aws-lc/blob/5cd1b450cdddafa080f40ba666df5fdfd162eaab/crypto/fipsmodule/cpucap/cpu_intel.c#L220-L225 and still allow the environment variable to set it?

Reply:
The RDRAND (ecx) bit is different.  The bit we're allowing to be set is the one that indicated "intel", which is the flag unset by this line: https://github.com/aws/aws-lc/blob/5cd1b450cdddafa080f40ba666df5fdfd162eaab/crypto/fipsmodule/cpucap/cpu_intel.c#L249